vital/algo/metadata_filter.h

```diff
@@ -0,0 +1,69 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+/**
+ * \file
+ * \brief Interface to abstract filter metadata algorithm
+ */
+
+#ifndef VITAL_ALGO_METADATA_FILTER_H
+#define VITAL_ALGO_METADATA_FILTER_H
+
+#include <vital/algo/algorithm.h>
+
+#include <vital/types/image_container.h>
+#include <vital/types/metadata.h>
+
+#include <vital/vital_config.h>
+
+namespace kwiver {
+
+namespace vital {
+
+namespace algo {
+
+/// Abstract base class for metadata filter algorithms.
+///
+/// This interface supports arrows/algorithms that modify image metadata.
+class VITAL_ALGO_EXPORT metadata_filter
+  : public kwiver::vital::algorithm_def< metadata_filter >
+{
+public:
+  /// Return the name of this algorithm.
+  static std::string static_type_name() { return "metadata_filter"; }
+
+  /// Return if this filter uses the image for filtering.
+  ///
+  /// This method returns whether the algorithm uses the image in its
+  /// implementation. It may be more efficient to not obtain the image if the
+  /// algorithm does not use it.
+  virtual bool uses_image() const = 0;
```

Comment:
Should this use "capabilities" instead?  It seems analogous to how capabilities are used in `video_input`, but I don't recall the exact motivation for capabilities.  Possibly just that we don't have to update all classes in a hierarchy to add a new capability.

Reply:
I'm... not sure. "Capabilities" is (AFAIK) normally used to ask "can you do X?". This doesn't quite fit that mold; rather, it's, "do you need X to do your work?". Possibly we could bludgeon it into that system by declaring a capability like CAN_DERIVE_METADATA_FROM_IMAGE_DATA?